### PR TITLE
Feature/switch retreat no troops

### PR DESCRIPTION
**Pull request by vmarkovtsev**
This is related to https://github.com/vcmi/vcmi/pull/167
Since this new switch breaks the save format, I send it as a separate PR

### CODE REVIEW

File: lib/CModHandler.h [hunk 1]
```diff
@@ -253,17 +253,20 @@ class DLL_LINKAGE CModHandler
...
 		int MAX_HEROES_ON_MAP_PER_PLAYER;
+		bool WINNING_HERO_WITH_NO_TROOPS_RETREATS;
 
 		template <typename Handler> void serialize(Handler &h, const int version)
 		{
 			h & data & CREEP_SIZE & WEEKLY_GROWTH & NEUTRAL_STACK_EXP & MAX_BUILDING_PER_TURN;
-			h & DWELLINGS_ACCUMULATE_CREATURES & ALL_CREATURES_GET_DOUBLE_MONTHS & MAX_HEROES_AVAILABLE_PER_PLAYER & MAX_HEROES_ON_MAP_PER_PLAYER;
+			h & DWELLINGS_ACCUMULATE_CREATURES & ALL_CREATURES_GET_DOUBLE_MONTHS &
+			MAX_HEROES_AVAILABLE_PER_PLAYER & MAX_HEROES_ON_MAP_PER_PLAYER &
+			WINNING_HERO_WITH_NO_TROOPS_RETREATS;
```
Comment (alexvins) [inline lib/CModHandler.h]: No need to break saves. just make if false for lower version.

Comment (vmarkovtsev) [inline lib/CModHandler.h]: Can you give an example? Where is the version, how to use it in serialization?..

Comment (alexvins) [inline lib/CModHandler.h]: "const int version" parameter is current used version. 

in Connection.h there are definitions of supported versions:

```cpp
const ui32 version = 755;
const ui32 minSupportedVersion = 753;
```

change it to 

```cpp
const ui32 version = 756;
const ui32 minSupportedVersion = 753;
```

and in serialization

```cpp
if(version >= 756) //save format backward compatibility
{
	h & WINNING_HERO_WITH_NO_TROOPS_RETREATS;
}
else if(!h.saving)
{
	WINNING_HERO_WITH_NO_TROOPS_RETREATS = false;
}
```

Comment (vmarkovtsev) [inline lib/CModHandler.h]: Thanks!

**Comment by ArseniyShestakov** [comment]
You can as well do same thing with #162 if you want it merged before save format breakage.

File: lib/filesystem/AdapterLoaders.h [hunk 2]
```diff
@@ -59,14 +59,14 @@ class DLL_LINKAGE CFilesystemList : public ISimpleResourceLoader
...
-        //class is not copyable 
-        return *this; 
+	CFilesystemList(CFilesystemList &)
+    {
+		//class is not copyable
+    }
+    CFilesystemList &operator=(CFilesystemList &)
+    {
+        //class is not copyable
+        return *this;
```
Comment (ArseniyShestakov) [inline lib/filesystem/AdapterLoaders.h]: Some mix of tabs and spaced there.

**Comment by vmarkovtsev** [comment]
Added new format version and worked on @ArseniyShestakov hint

**Pull request merged**